compiler/src/dmd/lexer.d

```diff
@@ -1784,9 +1866,17 @@ class Lexer
     *   D https://dlang.org/spec/lex.html#double_quoted_strings
     *   ImportC C11 6.4.5
     */
-    private void escapeStringConstant(Token* t)
+    private void escapeStringConstant(Token* t, bool supportInterpolation = false)
```

Comment:
It is probably better to split this into two functions (with a bit of repetition) instead. Should help with compilation speed.

Reply:
Do you have measurements of compilation speed difference?

Of course, such internal implementation details can be changed at any time.